.github/workflows/enteprise.yml

```diff
@@ -173,62 +175,52 @@ jobs:
+          BOT_GITHUB_TOKEN: ${{ secrets.BOT_GITHUB_TOKEN }}
+
+  ec2-manual:
+    name: ${{ matrix.molecule_scenario.desc }}
+    if: github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        molecule_scenario:
+          - name: default
+            vars: vars-ubuntu18.yml
+            desc: EC2 ACS 7.1 (Ubuntu 18.04)
+          - name: default
+            vars: vars-ubuntu20-72.yml
+            desc: EC2 ACS 7.2 (Ubuntu 20.04)
+    env:
+      AWS_REGION: eu-west-1
+      MOLECULE_IT_AWS_VPC_SUBNET_ID: subnet-6bdd4223
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
         with:
-          sanitize: true
-          max-length: 20
-
-      - name: Setup workspace
-        uses: ./.github/actions/setup-workspace
-
-      - name: Generate unique id
-        run: echo "MOLECULE_IT_ID=$(date +%s%4N)" >> $GITHUB_ENV
-
-      - name: Set nexus credentials only when required
-        if: ${{ ! contains( matrix.molecule_scenario.desc, 'community' ) }}
-        run: |
-          echo "NEXUS_USERNAME=${{ secrets.nexus_username }}" >> $GITHUB_ENV
-          echo "NEXUS_PASSWORD=${{ secrets.nexus_password }}" >> $GITHUB_ENV
-
-      - name: Run tests
-        timeout-minutes: 120
-        env:
-          MOLECULE_IT_SCENARIO: ${{ matrix.molecule_scenario.name }}
-          MOLECULE_IT_CONFIG: ${{ matrix.molecule_scenario.vars }}
-          AWS_ACCESS_KEY_ID: ${{ secrets.aws_access_key_id }}
-          AWS_SECRET_ACCESS_KEY: ${{ secrets.aws_secret_access_key }}
-          CLONE_GITHUB_TOKEN: ${{ secrets.BOT_GITHUB_TOKEN }}
-        run: ./tests/molecule_it/script.sh verify
+          # avoid auth conflict when cloning DTAS during verify step
+          persist-credentials: false
 
-      - name: Cleanup tests
-        if: always()
-        timeout-minutes: 65
-        env:
-          MOLECULE_IT_SCENARIO: ${{ matrix.molecule_scenario.name }}
-          MOLECULE_IT_CONFIG: ${{ matrix.molecule_scenario.vars }}
-          AWS_ACCESS_KEY_ID: ${{ secrets.aws_access_key_id }}
-          AWS_SECRET_ACCESS_KEY: ${{ secrets.aws_secret_access_key }}
-        uses: nick-fields/retry@v3
+      - name: Run tests manaully
+        uses: ./.github/actions/ec2
+        timeout-minutes: 185
         with:
-          timeout_minutes: 30
-          max_attempts: 3
-          retry_wait_seconds: 60
-          command: ./tests/molecule_it/script.sh destroy
+          matrix_name: ${{ matrix.molecule_scenario.name }}
+          matrix_vars: ${{ matrix.molecule_scenario.vars }}
+          matrix_desc: ${{ matrix.molecule_scenario.desc }}
+          nexus_username: ${{ secrets.NEXUS_USERNAME }}
+          nexus_password: ${{ secrets.NEXUS_PASSWORD }}
+          aws_access_key_id: ${{ secrets.aws_access_key_id }}
+          aws_secret_access_key: ${{ secrets.aws_secret_access_key }}
+          BOT_GITHUB_TOKEN: ${{ secrets.BOT_GITHUB_TOKEN }}
```

Comment:
since this action is not really meant to be reusable outside here you can avoid defining those inputs and use `${{ secrets.something }}` directly in the composite action

Reply:
Actions dont have access to secrets, reusable workflows do.

Reply:
https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#accessing-your-secrets